tests/test_extractors.py

```diff
@@ -91,3 +91,15 @@ def test_extract_from_directory(self, extractor, tmp_path, file):
     def test_extract_from_nonsense(self, extractor):
         with pytest.raises(NotImplementedError):
             extractor.extract(8)
+
+    @pytest.mark.parametrize(
+        "source",
+        [
+            """
+                def foo():
+            """,
+        ],
+    )
+    def test_extract_with_syntax_error(self, extractor, file_path, file, caplog):
```

Comment:
**suggestion (testing):** Verify the content of the warning message in the test.

While the test currently checks that a SyntaxWarning is raised, it would be valuable to also verify the content of the warning message. This ensures that the warning provides meaningful information about the syntax error, including the file path and the nature of the syntax error.